src/main/java/io/excaliburfrc/robot/subsystems/Climber.java

```diff
@@ -11,20 +22,55 @@
 import java.util.function.DoubleSupplier;
 
 public class Climber extends SubsystemBase implements AutoCloseable {
-  private final DoubleSolenoid anglerPiston =
+  private final DoubleSolenoid piston =
       new DoubleSolenoid(
           PneumaticsModuleType.CTREPCM,
           ClimberConstants.FORWARD_CHANNEL,
           ClimberConstants.REVERSE_CHANNEL);
-  private final CANSparkMax motor =
-      new CANSparkMax(ClimberConstants.CLIMBER_SPARKMAX, MotorType.kBrushless);
-  private final DigitalInput sensor = new DigitalInput(ClimberConstants.SENSOR_CHANNEL);
+  private final CANSparkMax leftMotor =
+      new CANSparkMax(ClimberConstants.LEFT_MOTOR_ID, MotorType.kBrushless);
+  private final CANSparkMax rightMotor =
+      new CANSparkMax(ClimberConstants.RIGHT_MOTOR_ID, MotorType.kBrushless);
+  private final RelativeEncoder leftEncoder = leftMotor.getEncoder();
+  private final SparkMaxPIDController leftController = leftMotor.getPIDController();
+  private final RelativeEncoder rightEncoder = rightMotor.getEncoder();
+  private final SparkMaxPIDController rightController = rightMotor.getPIDController();
+
+  private ElevatorFeedforward feedforward;
+
+  private final TrapezoidProfile.State trapezoidGoal = new TrapezoidProfile.State(0, 0);
+  private final TrapezoidProfile.State trapezoidInit = new TrapezoidProfile.State(0, 0);
+
+  public Climber() {
+    ValidateREVCAN(
+        // reset factory settings
+        leftMotor.restoreFactoryDefaults(),
+        rightMotor.restoreFactoryDefaults(),
+        // set the motors to brake mode
+        leftMotor.setIdleMode(CANSparkMax.IdleMode.kBrake),
+        rightMotor.setIdleMode(CANSparkMax.IdleMode.kBrake),
+        leftMotor.setPeriodicFramePeriod(PeriodicFrame.kStatus0, MAXIMAL_FRAME_PERIOD),
+        leftMotor.setPeriodicFramePeriod(PeriodicFrame.kStatus1, MAXIMAL_FRAME_PERIOD),
+        leftMotor.setPeriodicFramePeriod(PeriodicFrame.kStatus2, minimal_FRAME_PERIOD),
+        rightMotor.setPeriodicFramePeriod(PeriodicFrame.kStatus0, MAXIMAL_FRAME_PERIOD),
+        rightMotor.setPeriodicFramePeriod(PeriodicFrame.kStatus1, MAXIMAL_FRAME_PERIOD),
+        rightMotor.setPeriodicFramePeriod(PeriodicFrame.kStatus2, minimal_FRAME_PERIOD),
+        // set up PID parameters
+        leftController.setFeedbackDevice(leftEncoder),
+        leftController.setP(kP),
+        leftController.setI(kI),
+        leftController.setD(kD),
+        rightController.setFeedbackDevice(rightEncoder),
+        rightController.setP(kP),
+        rightController.setI(kI),
+        rightController.setD(kD));
```

Comment:
this PR is big enough as is, maybe the motor controller config should be split to a different PR (possibly done by someone else)?
That way, this PR can focus on adding the profiled control.